src/dragAndSnapManager.ts

```diff
@@ -0,0 +1,170 @@
+import {Position} from './enums/positionEnum';
+import {getClientX, getClientY, makeStyleString} from './utils';
+import {GuiClientRect} from './types';
+
+type OnPositionChangedCallback = (newPosition: Position) => void;
+enum DragEvents {
+  MouseDown = 'mousedown',
+  MouseMove = 'mousemove',
+  MouseUp = 'mouseup',
+
+  TouchStart = 'touchstart',
+  TouchMove = 'touchmove',
+  TouchEnd = 'touchend'
+}
+const DRAG_THROTTLE_MS = 30;
+
+export class DragAndSnapManager {
+  _eventManager: KalturaPlayerTypes.EventManager;
+  _logger: KalturaPlayerTypes.Logger;
+  _draggableContainer: HTMLDivElement | null = null;
+  _onPositionChanged: OnPositionChangedCallback;
+  _onStartDrag: () => void = () => {};
+  _onStopDrag: () => void = () => {};
+  _gueClientRec: GuiClientRect | null = null;
+  _currMousePos: {x: number; y: number} = {x: 0, y: 0};
+  _throttleWait: boolean = false;
+
+  constructor(eventManager: KalturaPlayerTypes.EventManager, logger: KalturaPlayerTypes.Logger, fn: OnPositionChangedCallback) {
+    this._eventManager = eventManager;
+    this._logger = logger;
+    this._onPositionChanged = fn;
+  }
+
+  // Drag and snap public API
+  public setDraggableContainer = (draggableContainer: HTMLDivElement) => {
+    this._draggableContainer = draggableContainer;
+    this._addListeners();
+  };
+
+  public setGuiClientRect = (gueClientRec: GuiClientRect) => {
+    this._gueClientRec = gueClientRec;
+  };
+
+  public onStartDrag = (fn: () => void) => {
+    this._onStartDrag = fn;
```

Comment:
Is it good this way? to get outside function references and keep them in the manager?
Looks a bit odd to me. For example. this means you cannot you the manager for more than one element at the same time. Maybe better to dispatch events from the manager

Reply:
we don't need have the manager for more than one element at the same time